pkg/cmd/operator/cleanupjob.go

```diff
@@ -118,11 +123,57 @@ func (o *CleanupJobOptions) Run(streams genericclioptions.IOStreams, cmd *cobra.
 	stopCh := signals.StopChannel()
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
+
 	go func() {
 		<-stopCh
 		cancel()
 	}()
 
+	var errs []error
+
+	cleanupErr := o.runCleanup(ctx)
+	if cleanupErr != nil {
+		errs = append(errs, fmt.Errorf("can't run the cleanup: %w", cleanupErr))
+	} else {
+		klog.InfoS("Node cleanup finished successfully")
+	}
+
+	select {
+	case <-stopCh:
+		if cleanupErr == nil {
+			break
+		}
+
+		klog.InfoS("Stopping any ongoing cleanup due to stop signal")
+		stopCleanupCtx, stopCleanupCtxCancel := context.WithTimeout(context.Background(), stopCleanupOperationTimeout)
+		defer stopCleanupCtxCancel()
+
+		wait.UntilWithContext(stopCleanupCtx, func(ctx context.Context) {
+			err := o.scyllaClient.StopCleanup(ctx, o.NodeAddress)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("can't stop the cleanup: %w", err))
+			} else {
+				stopCleanupCtxCancel()
+			}
+		}, time.Second)
+	default:
+	}
+
+	err := apierrors.NewAggregate(errs)
+	if err != nil {
+		return fmt.Errorf("can't clean up: %w", err)
+	}
+
+	return nil
+}
+
+func (o *CleanupJobOptions) runCleanup(ctx context.Context) error {
+	klog.InfoS("Stopping any ongoing cleanup")
+	err := o.scyllaClient.StopCleanup(ctx, o.NodeAddress)
```

Comment:
just to make sure: if there are no existing cleanups in progress, does this still return nil?

Reply:
yes:
```
root@basic-us-east-1-us-east-1a-0:/# curl -v -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' 'http://127.0.0.1:10000/compaction_manager/stop_compaction?type=CLEANUP'
*   Trying 127.0.0.1:10000...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 10000 (#0)
> POST /compaction_manager/stop_compaction?type=CLEANUP HTTP/1.1
> Host: 127.0.0.1:10000
> User-Agent: curl/7.68.0
> Content-Type: application/json
> Accept: application/json
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Length: 0
< Content-Type: application/json
< Date: Thu, 24 Aug 2023 20:04:28 GMT
< Server: Seastar httpd
< 
* Connection #0 to host 127.0.0.1 left intact
```